File: src/Starfield2026.MiniToolbox/src/MiniToolbox.Core/Bntx/BinaryData/Offset.cs
namespace MiniToolbox.Core.Bntx;

public class Offset
{
	public BinaryDataWriter Writer { get; private set; }

	public uint Position { get; private set; }

	public Offset(BinaryDataWriter writer)
	{
		Writer = writer;
		Position = (uint)Writer.Position;
		Writer.Position += 4L;
	}

	public void Satisfy()
	{
		Satisfy((int)Writer.Position);
	}

	public void Satisfy(int value)
	{
		uint num = (uint)Writer.Position;
		Writer.Position = Position;
		Writer.Write(value);
		Writer.Position = num;
	}
}

File: src/Starfield2026.MiniToolbox/src/MiniToolbox.Api/src/bntx/BinaryData/Offset.ts
import type { BinaryDataWriter } from './BinaryDataWriter.js';

/**
 * 1:1 port of Offset.cs
 * Reserves space for an offset value to be filled in later (write-only).
 */
export class Offset {
    readonly writer: BinaryDataWriter;
    readonly position: number;

    constructor(writer: BinaryDataWriter) {
        this.writer = writer;
        this.position = writer.position;
        writer.position += 4; // Reserve 4 bytes
    }

    satisfy(value?: number): void {
        const current = this.writer.position;
        this.writer.position = this.position;
        this.writer.writeInt32(value ?? this.writer.position);
        this.writer.position = current;
    }
}
